ui/plugin-system/src/components/PluginKindSelect/PluginKindSelect.tsx

```diff
@@ -31,13 +32,17 @@ export interface PluginKindSelectProps extends Omit<TextFieldProps, 'value' | 'o
  * when the user changes the plugin type (it fires at start for the default value.)
  */
 export const PluginKindSelect = forwardRef((props: PluginKindSelectProps, ref): ReactElement => {
-  const { pluginTypes, value: propValue, onChange, ...others } = props;
+  const { pluginTypes, value: propValue, onChange, filteredQueryPlugins, ...others } = props;
   const { data, isLoading } = useListPluginMetadata(pluginTypes);
 
-  const sortedData = useMemo(
-    () => data?.sort((a, b) => a.spec.display.name.localeCompare(b.spec.display.name)),
-    [data]
-  );
+  const sortedData = useMemo(() => {
+    if (!filteredQueryPlugins?.length || !data?.map((i) => i.spec.name).some((i) => filteredQueryPlugins.includes(i)))
+      return data?.sort((a, b) => a.spec.display.name.localeCompare(b.spec.display.name));
```

Comment:
Let's add some brackets :)
```suggestion
{
      return data?.sort((a, b) => a.spec.display.name.localeCompare(b.spec.display.name));
      }
```